packages/ui/src/components/TimeInputV2/index.tsx

```diff
@@ -0,0 +1,511 @@
+                  onClick={event => {
+                    event.stopPropagation()
+                  }}
+                  ref={computedRef()}
+                  role="spinbutton"
+                  aria-valuemax={computeMaxValue()}
+                  aria-valuemin={type === 'h' && timeFormat === 12 ? 1 : 0}
+                  aria-valuenow={Number.parseInt(time[type], 10)}
+                  onChange={event => {
+                    if (!readOnly && !disabled) {
+                      const key = getLastTypedChar(
+                        event.target.value,
+                        time[type],
+                      )
+                      if (isNumber(key)) handleChange(type, key)
+                    }
+                  }}
+                  onKeyDown={event => {
+                    if (!readOnly && !disabled) {
+                      if (event.key === 'ArrowUp') {
+                        event.preventDefault()
+                        handleIncrease(type)
+                      } else if (event.key === 'ArrowDown') {
+                        event.preventDefault()
+                        handleDecrease(type)
+                      } else if (event.key === 'ArrowLeft') {
+                        event.preventDefault()
+                        handlePrevious(type)
+                      } else if (event.key === 'ArrowRight') {
+                        event.preventDefault()
+                        handleNext(type)
+                      }
+                    }
+                  }}
+                  autoFocus={autoFocus && type === 'h'}
+                />
+                {type === 's' ? null : (
+                  <CustomText
+                    as="span"
+                    variant="body"
+                    prominence="default"
+                    sentiment="neutral"
+                  >
+                    :
+                  </CustomText>
+                )}
+              </Stack>
+            )
+          })}
+          {timeFormat === 12 ? (
+            <Input
+              value={time.period?.toUpperCase()}
+              placeholder={placeholder.period ?? 'AM'}
+              data-size={size}
+              data-period
+              readOnly={readOnly}
+              disabled={disabled}
+              aria-label="am-pm-input"
+              data-testid="am-pm-input"
+              onChange={event => {
+                if (!readOnly && !disabled) {
+                  const key = event.target.value.slice(-1)
+                  if (isAOrP(key)) handleChange('period', key)
+                }
+              }}
+              onKeyDown={event => {
+                if (!readOnly && !disabled) {
+                  if (event.key === 'ArrowUp' || event.key === 'ArrowDown') {
+                    event.preventDefault()
+                    const newTime = {
+                      ...time,
+                      period:
+                        time.period === 'am' ? 'pm' : ('am' as 'am' | 'pm'),
```

Comment:
```suggestion
                        time.period === 'am' ? 'pm' : 'am',
```
You shouldn't need that.